### PR TITLE
Tomas Jablonskis - UI Design Patterns Challenge

### PR DESCRIPTION
**Pull request by tomasjablonskis**

### CODE REVIEW

File: MVP Module/Source/Feed UI/Views/Feed.storyboard [hunk 1]
```diff
@@ -14,19 +14,41 @@
                         <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                         <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
-                        <view key="tableHeaderView" contentMode="scaleToFill" id="clu-9B-M8g">
-                            <rect key="frame" x="0.0" y="0.0" width="414" height="16"/>
+                        <view key="tableHeaderView" contentMode="scaleToFill" id="xk6-8z-f3Y" customClass="ErrorView" customModule="MVVM" customModuleProvider="target">
```
Comment (danillahtin) [inline MVP Module/Source/Feed UI/Views/Feed.storyboard:17]: This storyboard is in the MVP solution, so the `customModule` should be set to MVP.

```suggestion
                        <view key="tableHeaderView" contentMode="scaleToFill" id="xk6-8z-f3Y" customClass="ErrorView" customModule="MVP" customModuleProvider="target">
```

File: MVVM Module/Source/Feed Presentation/FeedViewModel.swift [hunk 2]
```diff
@@ -20,12 +20,18 @@ final class FeedViewModel {
 
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onErrorStateChanged: Observer<String?>?
```
Comment (danillahtin) [inline MVVM Module/Source/Feed Presentation/FeedViewModel.swift:23]: Suggestion:
`onErrorStateChanged` can be named `onErrorStateChange` (without `d` at the end) to be consistent with `onLoadingStateChange`
```suggestion
	var onErrorStateChange: Observer<String?>?
```

**Review by danillahtin — COMMENTED** [review]
Hi @tomasjablonskis, you almost passed the challenge!

This is Danil, I am an instructor at Essential Developer.

Please, check my review comments and let me know once you're done so I can review it again. 

Thank you 🙂

**Comment by tomasjablonskis** [comment]
Hi, @danillahtin

Finally found some free time to look at the issues and fix them. PR ready for re-review.

**Review by danillahtin — COMMENTED** [review]
Hi @tomasjablonskis,

Congratulations, you've passed the challenge and got your third belt stripe ⚪️③!

✅ All tests are passing
✅ Your code is clean and handles edge cases
✅ The UI and animations look and behave as expected
✅ Light and Dark mode support
✅ Added an extra language

**[Update your status](https://iosleadessentials.slack.com/archives/CTAR0LVMM/p1629389971188200) in the community now with the following emoji to celebrate your achievement: `:white-belt-3:`**

Keep up the great work! There's one last challenge ahead. Remember, we're here to help you progress and get the most out of this program.

See you in the community and in the final challenge 🙂
